src/topics/topicActions.js

```diff
@@ -40,25 +40,22 @@ export const fetchTopicsForStream = (narrow: Narrow) => async (
   dispatch(fetchTopics(stream.stream_id));
 };
 
-export const deleteMessagesForTopic = (streamName: string, topic: string) => async (
+export const deleteMessagesForTopic = (streamId: number, topic: string) => async (
   dispatch: Dispatch,
   getState: GetState,
 ) => {
   const state = getState();
   const outbox = getOutbox(state);
+  const currentStream = getStreamsById(state).get(streamId);
+
   outbox.forEach((outboxMessage: Outbox) => {
     if (
       outboxMessage.type === 'stream'
-      && streamNameOfStreamMessage(outboxMessage) === streamName
+      && streamNameOfStreamMessage(outboxMessage) === currentStream?.name
```

Comment:
Let's add a TODO comment here that mentions #3998 and #3918. That will help us find this spot and clean it up to use the stream ID too, once we've added stream IDs to our outbox messages.